### DISCUSSION TITLE
Logo/Icon

### DISCUSSION THREAD

**Discussion by tinydogio-joshua** [counterfact/api-simulator#260] [Ideas]
You need a logo/icon. 😀

**Comment by pmcelhaney** [comment]
I opened up the issues list to create this exact issue and this was a the top. :)

Here are some straw men. I'm not very good at this. 

![image](https://user-images.githubusercontent.com/51504/194766434-4b550d78-388f-4e8b-a4eb-aeb06ed73290.png)

**Comment by pmcelhaney** [comment]
Google image search for "[CF Logo](https://www.google.com/search?q=cf+logo)" has a bunch of ideas.

![image](https://user-images.githubusercontent.com/51504/194766658-40f1d627-f9f1-4388-a904-24d41ba9a988.png)

**Comment by pmcelhaney** [comment]
![image](https://user-images.githubusercontent.com/51504/194767241-c27f42b8-10c7-4117-87c6-b792c9948c8b.png)

![image](https://user-images.githubusercontent.com/51504/194767325-8fadf70b-e1bf-457a-9e6a-b19c354a44eb.png)

**Reply by pmcelhaney** [reply]
With Menlo font

![image](https://user-images.githubusercontent.com/51504/194767984-f1df7dd5-46b3-476c-9d65-7e16d1e5989f.png)


![image](https://user-images.githubusercontent.com/51504/194768044-c00aff7e-0d98-47d3-abd7-4031243b56d8.png)

**Reply by pmcelhaney** [reply]
Roboto Mono (no licensing issues)

![image](https://user-images.githubusercontent.com/51504/194768622-5120f0fe-42c3-454b-8a3a-a375f5a627bc.png)

![image](https://user-images.githubusercontent.com/51504/194768635-19ef42cd-be17-46b6-9d99-4a15a8eae3ec.png)

**Comment by tinydogio-joshua** [comment]
> Looks like I may be a little late to this, but if you'd still like for me to take a stab at it, please answer the following. I can also use the above as a starting point if you'd like. 😀
---
## Logo Design Questionnaire:

1. Please list about 5 products or services that are similar to (or in the same family) as the product or services you are creating. Provide URL's if available.

2. Please list at least 5 *logos* that you like the look of, and what elements you like about each. They do not have to be the same type of product or service you wish to build.
_Examples: Cute Mascot, Colors, Symmetry

3. Please list some colors you would like associated with your product or service.
_Please note that they may not all be used, so if their is one you absolutely want please note it._

4. Do you have a target audience / demographic? If so, what is it?

5. If you had to pick a soundtrack for your product or service, which bands and type of music would you associate with it?
_It may help to try and imagine someone browsing your product or service in a movie, or the website being shown in a commercial._

6. How do you want people to feel when they view and use your product or service?
_Examples: Energized, Somber, Reflective_

7. What is the primary focus of your product? Meaning what do you want people to do with your product or service?
_Examples: Learn Something, Buy a Product, Engage in a Discussion, Build Something._

**Reply by pmcelhaney** [reply]
**Please list about 5 products or services that are similar to (or in the same family) as the product or services you are creating. Provide URL's if available.**

- https://mswjs.io/
- https://swagger.io/
- https://mockoon.com/
- https://enhance.dev/
- https://koajs.com/

**Please list at least 5 logos that you like the look of, and what elements you like about each. They do not have to be the same type of product or service you wish to build.**

![Doordash](https://user-images.githubusercontent.com/51504/196533896-927cf206-c00a-4538-92ed-69141a84ee67.png)

Doordash. It's simple and it evokes speed. 



<img src="https://user-images.githubusercontent.com/51504/196535135-dea9efc9-286e-4944-b219-80cca1688e4f.png" height="100">

I like the negative space and the fact that something is plugging into something else.


<img src="https://user-images.githubusercontent.com/51504/196537229-0612798c-b1a8-4dcf-92be-19fa09d6d6df.png" height="100">

I like the fact that four simple triangles can also be an N and also be a lightning bolt. 



<img src="https://user-images.githubusercontent.com/51504/196537698-e27d1a48-cea8-4c40-afbe-eedebead5b14.png" height="100">

The masks represent the idea of mocking. 


<img src="https://user-images.githubusercontent.com/51504/196540326-cf6ae6d1-12ae-46b9-b1aa-389e627720df.png" height="100">

If you wield this tool you're a superhero. 


<img src="https://user-images.githubusercontent.com/51504/196541343-fb66bb42-80c2-4fee-b41d-6a76267a2443.png" height="100">

I like the idea of the F turning around to fit into the C. Two things plugging into one another and one of them is flipped around (counter, dependency inversion, fake). I just found this one, but as you can see above I was already playing with the idea.


<img src="https://user-images.githubusercontent.com/51504/196542688-eeb43eaf-dbab-4fdb-b504-9f08ada059cd.png" height="100">

I was thinking about hexagonal architecture, searched for "hexagon c logo", and found this one. I like that it kind of evokes Angular and HTML 5. If you like it maybe I can just [buy](https://designbundles.net/yopie/2105558-letter-c-hexagon-logo-template) it. The fact that a 2D projection of a cube is a hexagon is pretty interesting. 


<img src="https://user-images.githubusercontent.com/51504/196549189-95f9f7bc-d25f-4f0f-b1c1-94437712b63a.png" height="100">

That last one kind of looked like a wrench, which implies building. I Googled "wrench hexagon logo" and found [this](https://www.vectorstock.com/royalty-free-vector/simple-hexagon-wrench-logo-icon-vector-29557090). The fact that the smaller hexagon isn't in the center kind of bothers me. 

_Edit: turns out I like the wrench thing because subconsciously I was remembering the [OpenAPI Tools](https://github.com/OpenAPITools) logo._ 

**Please list some colors you would like associated with your product or service.**

I played around with the orange from the HTML 5 logo because I thought that might feel familiar to front end devs. But really I have no idea. 


**Do you have a target audience / demographic? If so, what is it?**

Front-end devs.



**If you had to pick a soundtrack for your product or service, which bands and type of music would you associate with it?**

These songs popped into my head. 

- [Stressed Out by 21 Pilots](https://www.youtube.com/watch?v=pXRviuL6vMY)
- [What's Up Danger by Blackaway with Black Caviar](https://www.youtube.com/watch?v=Y88LVU7MAe4)
- [Thrift Shop by Macklemore & Ryan Lewis](https://www.youtube.com/watch?v=QK8mJJJvaes)
- [Cigarettes on Patios by BabyJake & 24k Golden](https://www.youtube.com/watch?v=WHZlQPi9wcQ)
- [Cooler than Me by Mike Posner](https://www.youtube.com/watch?v=mqWq_48LxWQ)
- [Parents Just Don't Understand by DJ Jazzy Jeff & The Fresh Prince](https://www.youtube.com/watch?v=jW3PFC86UNI) 

My post hoc explanation is they tell a story about how life is hard, but I have a different perspective, I'm not gonna let the hardship bother me, I'm gonna bounce along and have fun. 

**How do you want people to feel when they view and use your product or service?**

Unblocked, creative, productive


**What is the primary focus of your product? Meaning what do you want people to do with your product or service?**

Go be a front end engineer and build the front end and stop worrying about the back end and all the complexity and long feedback loops that entails.

**Reply by pmcelhaney** [reply]
LOL whoops https://github.com/OpenAPITools

**Reply by tinydogio-joshua** [reply]
Hoping to get to this soon. 😀

**Reply by tinydogio-joshua** [reply]
Thank you for filling this out.

**Reply by pmcelhaney** [reply]
No no no thank _you!!!_

**Reply by tinydogio-joshua** [reply]
I have a few rough sketches to show you tomorrow if you have some time. Just let me know.

**Comment by tinydogio-joshua** [comment]
> VERY ROUGH SKETCHES BELOW. WE'D LOVE SOME FEEDBACK OR ADDITIONAL OPTIONS. 🙏

Note that these aren't meant to be the symbol/bug. I'd use them as a starting point for a bigger logo.

![Counterfact Logo Sketches](https://user-images.githubusercontent.com/18430941/197561021-e6c2e1cd-fa2a-4681-8914-135685f8f65a.png)

**Comment by yuriykatko** [comment]
gave DALL-E a go (don't mind the actual label)
![DALL·E 2022-10-24 11 53 19](https://user-images.githubusercontent.com/8705506/197570507-694b3cf3-3373-49fe-8dc2-3853ffde7d29.png)

**Reply by tinydogio-joshua** [reply]
Not bad. 😀

**Reply by tinydogio-joshua** [reply]
@pmcelhaney I am hoping to spend some time on this soon. Do you have any preference for a starting point?

**Reply by pmcelhaney** [reply]
@tinydogio-joshua Sorry, just saw this. The black one speaks to me most. 

I want people to understand that Counterfact is the equivalent of paper prototypes or wireframes on the server side. People think it's duplicate work, but it's not. It's a tiny amount of work compared to building the real back end. It adds zero value to the shipped product and a ton of value to the process of building the product.

**Comment by tinydogio-joshua** [comment]
What do you think of something like this? It was suggested to add a few small details. I also thought adding some accent colors (which can be adjusted) would be nice.

<img width="1440" alt="Screenshot 2023-12-05 at 2 42 08 pm" src="https://github.com/pmcelhaney/counterfact/assets/18430941/4771f048-8903-45ef-8e43-52a87a54bd5d">

**Reply by tinydogio-joshua** [reply]
I'll add a non "bug" version with text once I know this is ok.

**Reply by tinydogio-joshua** [reply]
Trying out some text treatments next to it. This is with Open Sans.

<img width="1440" alt="Screenshot 2023-12-05 at 3 07 24 pm" src="https://github.com/pmcelhaney/counterfact/assets/18430941/d0d63a5c-55d6-4a26-b49c-e58fe3c1738f">

**Reply by pmcelhaney** [reply]
I like the background with the accent colors. Although it looks like an app icon (as in a GUI app, not a server). Looks good but I think it's the wrong direction.

**Reply by tinydogio-joshua** [reply]
ok. I will leave this here in case I ever need to come back to it. Shows it more in context of other graphics.

<img width="1440" alt="Screenshot 2023-12-05 at 3 29 40 pm" src="https://github.com/pmcelhaney/counterfact/assets/18430941/b381b617-fd2e-4127-a59a-85bbbf5866cc">

**Comment by pmcelhaney** [comment]
Thinking about the wordplay, I worked with ChatGPT and came up with this concept.

![image](https://github.com/pmcelhaney/counterfact/assets/51504/90f3f8c4-d5c8-41fb-afeb-6f57a406311e)

I like the playfulness of the wink. It says, yeah, I'm "working hard" to get all this stuff done.

**Reply by pmcelhaney** [reply]
That's supposed to be a coin, BTW.

![image](https://github.com/pmcelhaney/counterfact/assets/51504/f37f9b79-566e-4e22-b9fc-8f4d092732b9)

**Reply by tinydogio-joshua** [reply]
I'll try to make something more fun and less "app icon" like. I'm more of a UI designer, so my designs tend to be more aligned that way. I'm not much of an illustrator (which is what I'm getting from the above). I'll see what I can do, but they probably won't be as good as these.

**Reply by tinydogio-joshua** [reply]
This is how far I got today. I. need to take off. I will provide another update tomorrow with text treatments.

<img width="1440" alt="Screenshot 2023-12-05 at 5 04 57 pm" src="https://github.com/pmcelhaney/counterfact/assets/18430941/27986f8d-9554-4533-8068-12d49fb68c44">

**Reply by tinydogio-joshua** [reply]
What's your favorite color @pmcelhaney ? In general... don't worry about the colors above. Thinking of switching it around a bit; or at least messing around with other colors.

**Reply by pmcelhaney** [reply]
#2f94e3 or thereabouts

**Comment by pmcelhaney** [comment]
I had ChatGPT write some marketing copy for me, then generate logos inspired by that marketing copy. I like the top left here. 

![image](https://github.com/pmcelhaney/counterfact/assets/51504/f28c7ca6-c87a-4b7f-87bb-e3495d3fbd56)

**Reply by pmcelhaney** [reply]
I'm polling a few more people. Will select a winner in a few hours. As of now bottom left is winning.

**Comment by pmcelhaney** [comment]
![image](https://github.com/pmcelhaney/counterfact/assets/51504/b2493311-5741-4480-ad7a-d28ac9738731)

Created this SVG by hand. It needs to be tweaked a little bit. 

```svg
<svg width="300" height="300" xmlns="http://www.w3.org/2000/svg">
    <polygon points="150,150 63.3974596216,300 236.602540378, 300 " fill="#0071b5" stroke="#0071b5"/>
    <polygon points="150,150 236.602540378,300 300,150 " fill="#f58e00" stroke="#f58e00"/>
    <polygon points="150,150 236.602540378,0 300,150 " fill="#0071b5" stroke="#0071b5"/>
    <polygon points="150,150 236.602540378,0, 63.3974596216,0" fill="#0071b5" stroke="#0071b5"/>
    <polygon points="150,150 0,150, 63.3974596216,0" fill="#0071b5" stroke="#0071b5"/>
    <polygon points="150,150 0,150, 63.3974596216,300" fill="#0071b5" stroke="#0071b5"/>
</svg>
```

**Reply by dethell** [reply]
I like this. Simple and I like the slice of pie theme. Brings to mind that I'm swapping out a piece of the polygon.

**Reply by pmcelhaney** [reply]
Great! That's exactly what I was going for!

**Reply by pmcelhaney** [reply]
Here's a script to generate the hexagon. 

```js
const xmlns = "http://www.w3.org/2000/svg";
const width = 200;
const height = 200;
const size = 50; // Adjusted size for better fit
const centerX = width / 2;
const centerY = height / 2;
const angle = Math.PI / 3; // 60 degrees in radians

const svg = document.createElementNS(xmlns, "svg");
svg.setAttributeNS(null, "width", width);
svg.setAttributeNS(null, "height", height);
svg.setAttributeNS(null, "viewBox", `0 0 ${width} ${height}`);

function createTriangle(x, y, rotation, offset = 6) {
  const points = [[x, y]];
  for (let i = 0; i < 2; i++) {
    const theta = i * (2 * Math.PI / 6) + rotation; // Corrected divisor to 6
    const px = x + size * Math.cos(theta);
    const py = y + size * Math.sin(theta);
    points.push([px, py]);
  }
  
  const offsetRotation = rotation + Math.PI / 6;
  const offsetPoints = points.map(([xx, yy]) => [xx + Math.cos(offsetRotation) * offset , yy + Math.sin(offsetRotation) * offset]);
  
  const polygon = document.createElementNS(xmlns, "polygon");
  polygon.setAttributeNS(null, "points", offsetPoints.join(" "));
  polygon.setAttributeNS(null, "fill", rotation === 0 ? "#f58e00" : "#0071b5");

  svg.appendChild(polygon);
}

for (let i = 0; i < 6; i++) {
  const rotation = i * angle;
  createTriangle(centerX, centerY, rotation);
}

document.body.appendChild(svg);
```

**Reply by pmcelhaney** [reply]
Finally, an SVG file. It could still be improved by someone who has access to a good editor and knows how to use it. 

![counterfact](https://github.com/pmcelhaney/counterfact/assets/51504/88c21e9d-942d-44a1-ada3-f1b063d516f6)
